src/borg/platform/posix.pyx

```diff
@@ -6,21 +6,32 @@ from functools import lru_cache
 
 from libc.errno cimport errno as c_errno
 
+from cpython.mem cimport PyMem_Free
+from libc.stddef cimport wchar_t
+
 cdef extern from "wchar.h":
-    cdef int wcswidth(const Py_UNICODE *str, size_t n)
+    # https://www.man7.org/linux/man-pages/man3/wcswidth.3.html
+    cdef int wcswidth(const wchar_t *s, size_t n)
+
+
+cdef extern from "Python.h":
+    # https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_AsWideCharString
+    wchar_t* PyUnicode_AsWideCharString(object, Py_ssize_t*) except NULL
```

Comment:
that is "part of the stable abi since python 3.7", so guess we can't use this in 1.1-maint.

Reply:
otoh, it says "new in 3.2". so, do we have it in 3.5 and 3.6 ?

Reply:
The API was added in 3.2, the ABI was frozen in 3.7. Do we need stable ABI?

Reply:
hmm, guess not. we bundle precompiled .c files into releases, but the c code is then compiled on the target system.

so guess it is good enough if the api is there.